File: src/popsalon/backend/src/Popsalon.Application/EntityViews/CustomerView.cs
// FICHIER GÉNÉRÉ AUTOMATIQUEMENT — forge generate
// Source : metadata/entities/Customer.yml

namespace Popsalon.Application.EntityViews;

/// <summary>Modèle de lecture (CQRS read side) pour l'entité Customer.</summary>
public sealed record CustomerView
{
    public Guid Id { get; init; }
    public string FirstName { get; init; } = "";
    public string LastName { get; init; } = "";
    public string FullName { get; init; } = "";
    public string? Email { get; init; }
    public string? Phone { get; init; }
}

File: src/popsalon/frontend/src/dataObjects/CustomerView.ts
// FICHIER GÉNÉRÉ AUTOMATIQUEMENT — forge generate
// Source : metadata/entities/Customer.yml

export interface CustomerView {
  id: string
  firstName: string
  lastName: string
  fullName: string
  email: string | null
  phone: string | null
}

export interface CreateCustomerModel {
  firstName: string
  lastName: string
  email: string | null
  phone: string | null
}

export interface UpdateCustomerModel {
  id: string
  firstName: string
  lastName: string
  email: string | null
  phone: string | null
}
